src/modules/formBuilder/components/formTree/FormTreeLabel.tsx

```diff
@@ -28,42 +32,56 @@ export interface FormTreeLabelProps
   itemId: string;
 }
 
+const dependentLabelMap: Record<string, string> = {
+  autofillDependents: 'Items with autofill condition(s)',
+  enableWhenDependents: 'Items with visibility condition(s)',
+  boundDependents: 'Items with min/max bound(s)',
+};
+
 const FormTreeLabel: React.FC<FormTreeLabelProps> = ({
   id,
   itemId,
   label,
   disabled,
   children,
 }) => {
-  const { openFormItemEditor } = React.useContext(FormTreeContext);
+  const { openFormItemEditor, itemMap, rhfPathMap, expandItem } =
+    useContext(FormTreeContext);
 
   const { control } = useFormContext();
+
   const { isSubmitting } = useFormState({ control });
 
-  const { getLabelProps, publicAPI } = useTreeItem2({
+  const { getLabelProps } = useTreeItem2({
     id,
     itemId,
     children,
     label,
     disabled,
   });
 
-  // we could get this from the form instead of from the tree api
-  const treeItem = publicAPI.getItem(itemId);
-  const item = useMemo(() => getItemFromTree(treeItem), [treeItem]);
+  const item = useMemo(() => itemMap[itemId], [itemMap, itemId]);
 
   const displayAttrs = useMemo(
-    () => getItemDisplayAttrs(item.type),
-    [item.type]
+    () => getItemDisplayAttrs(item?.type),
+    [item?.type]
   );
 
   const labelProps = getLabelProps();
 
-  const { onReorder, canMoveUp, canMoveDown } = useReorderItem(
-    control,
-    itemId,
-    item
-  );
+  const { onReorder, onDelete, canMoveUp, canMoveDown } =
+    useUpdateFormStructure(
+      control,
+      itemId,
+      item,
+      itemMap,
+      rhfPathMap,
+      expandItem
+    );
+
+  const [itemDependents, setItemDependents] = useState<
+    ItemDependents | undefined
+  >(undefined);
```

Comment:
Suggest adding a comment to explain that this piece of state relates specifically to items that are blocking a deleting that is currently being attempted. It could be mistaken for just holding the items that are dependent on this item, at all times.